kratos/sources/model_part.cpp

```diff
@@ -1616,28 +1632,71 @@ void ModelPart::RemoveGeometryFromAllLevels(std::string GeometryName)
 ///@name Sub Model Parts
 ///@{
 
-ModelPart&  ModelPart::CreateSubModelPart(std::string const& NewSubModelPartName)
+ModelPart& ModelPart::CreateSubModelPart(std::string const& NewSubModelPartName)
 {
-    // Here a warning would be enough. To be disscussed. Pooyan.
-    KRATOS_ERROR_IF(mSubModelParts.find(NewSubModelPartName) != mSubModelParts.end())
-        << "There is an already existing sub model part with name \"" << NewSubModelPartName
-        << "\" in model part: \"" << Name() << "\"" << std::endl;
+    const auto sub_model_parts_list = SplitSubModelPartHierarchy(NewSubModelPartName);
+    if (sub_model_parts_list.size() == 1) {
+        // Here a warning would be enough. To be disscussed. Pooyan.
+        KRATOS_ERROR_IF(mSubModelParts.find(NewSubModelPartName) != mSubModelParts.end())
+            << "There is an already existing sub model part with name \"" << NewSubModelPartName
+            << "\" in model part: \"" << Name() << "\"" << std::endl;
+
+        ModelPart* praw = new ModelPart(NewSubModelPartName, this->mpVariablesList, this->GetModel());
+        Kratos::shared_ptr<ModelPart> p_model_part(praw); //we need to construct first a raw pointer
```

Comment:
Is this required ? first a raw and then a shared pointer ?

Reply:
pretty sure it is, @RiccardoRossi wrote this

Reply:
thats bcs `ModelPart` doesn't have a public constructor, and `unique_ptr` is not a `friend` of `ModelPart`, but `Model` is